src/ansys/mechanical/core/embedding/logger/__init__.py

```diff
@@ -0,0 +1,193 @@
+            Base directory to write log files to. This is only possible to set before the
+            Mechanical application is initialized.
+        to_stdout : bool, optional
+            Whether to write log messages to the standard output, which is the
+            command line. The default is ``True``.
+        """
+        # Setup the global log configuration.
+        cls.set_log_directory(directory)
+        cls.set_log_base_directory(base_directory)
+
+        # Setup the sink-specific log configuration, store to global state.
+        cls._store_stdout_sink_enabled(to_stdout)
+        file_sink_enabled = directory != None or base_directory != None
+        cls._store_file_sink_enabled(file_sink_enabled)
+
+        # Commit the sink-specific log configuration global state to the backend.
+        cls._commit_enabled_configuration()
+        cls.set_log_level(level)
+
+    @classmethod
+    def set_log_to_stdout(cls, value: bool) -> None:
+        """Configure logging to write to the stdout."""
+        cls._store_stdout_sink_enabled(value)
+        cls._commit_enabled_configuration()
+
+    @classmethod
+    def set_log_to_file(cls, value: bool) -> None:
+        """Configure logging to write to a file."""
+        cls._store_file_sink_enabled(value)
+        cls._commit_enabled_configuration()
+
+    @classmethod
+    def set_log_level(cls, level: int) -> None:
+        """Set the log level for all configured sinks."""
+        if len(LOGGING_SINKS) == 0:
+            raise Exception("No logging backend configured!")
+        cls._commit_level_configuration(level)
+
+    @classmethod
+    def set_log_directory(cls, value: str) -> None:
+        """Configure logging to write to a directory."""
+        if value == None:
+            return
+        _get_backend().set_directory(value)
+
+    @classmethod
+    def set_log_base_directory(cls, value: str) -> None:
+        """Configure logging to write in a time-stamped subfolder in the given directory."""
+        if value == None:
+            return
+        _get_backend().set_base_directory(value)
+
+    @classmethod
+    def _commit_level_configuration(cls, level: int) -> None:
+        for sink in LOGGING_SINKS:
+            _get_backend().set_log_level(level, sink)
+
+    @classmethod
+    def _commit_enabled_configuration(cls) -> None:
+        for sink in LOGGING_SINKS:
+            _get_backend().enable(sink)
+
+    @classmethod
+    def _store_stdout_sink_enabled(cls, value: bool) -> None:
+        if value:
+            LOGGING_SINKS.add(sinks.StandardSinks.CONSOLE)
+        else:
+            LOGGING_SINKS.discard(sinks.StandardSinks.CONSOLE)
+
+    @classmethod
+    def _store_file_sink_enabled(cls, value: bool) -> None:
+        if value:
+            LOGGING_SINKS.add(sinks.StandardSinks.STANDARD_LOG_FILE)
+        else:
+            LOGGING_SINKS.discard(sinks.StandardSinks.STANDARD_LOG_FILE)
+
+
+class Logger:
+    """Logger class for embedding."""
```

Comment:
```suggestion
    """Provides the ``Logger`` class for embedding."""
```
All class descriptions should start with a verb ending in an "s" or "es".